### PR TITLE
feat(ui/pages): search page

### LINKED ISSUE #18: Search Result Page (/search)
### What is this for?

website

### Feature

As a user, I want to be able to search and evaluate results from all communities on Answer Overflow to find the answers I'm looking for.

### Ideal solution or implementation

A search bar that takes you to a results page. The results page needs to have information such as:

- # Replies
- # Views
- Is Solved?
- Community

### Alternative solutions or implementations

_No response_

### Other context

View the existing search as a reference:

https://www.answeroverflow.com/search?q=How+do+I+turn+on+these+new+settings+for+my+help+channels%3F

### LINKED ISSUE #196: [Web] Text input component
### Purpose?

A text input component

### Concepts

N/A, just a text input

### Description

It should contain the appropriate props and event handlers, along with icons

### Extra Information

_No response_

### PR DESCRIPTION
**Pull request by JoltCode**
# Description

Adds a search page amongst other things (will update when ready with complete list)

Fixes #18 (issue)
Closes #196 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

N/A

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works 
- [x] I have updated .env.example if I added a new environment variable
- [x] My PR title follows the [semantic commits](https://www.conventionalcommits.org/en/v1.0.0/) style

### CODE REVIEW

File: packages/config/eslint-config-custom/next.js [hunk 1]
```diff
@@ -23,13 +23,7 @@ module.exports = {
...
   plugins: ["tailwindcss"],
   rules: {
-    "tailwindcss/no-custom-classname": [
-      "warn", // TODO: Renable this eventually, whitelist is not working
-      {
-        config: "./tailwind.config.cjs",
-        whitelist: ["scrollbar-hide", "dark"],
-      },
-    ],
+    "tailwindcss/no-custom-classname": "off",
```
Comment (RhysSullivan) [inline packages/config/eslint-config-custom/next.js:26]: do we want this off?

File: packages/ui/src/components/home/Home.tsx [hunk 2]
```diff
@@ -1,15 +1,15 @@
 import { ChevronDownIcon } from '@heroicons/react/24/outline';
 import { Navbar } from '../primitives/Navbar';
-import { BlobHome } from './BlobHome';
+// import { BlobHome } from "./BlobHome";
```
Comment (RhysSullivan) [inline packages/ui/src/components/home/Home.tsx:3]: delete comment

File: packages/ui/src/components/home/Home.tsx [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 import { QuestionAnswerArea } from './QuestionAnswerArea/QuestionAnswerArea';
 import { messageData } from './HomeMessages';
 
 export const Home = () => {
 	return (
-		<div className="absolute -z-50 h-screen w-screen overflow-hidden bg-ao-white bg-gradient-to-t from-[#10151c] to-[#354364] xl:from-transparent xl:to-transparent xl:dark:bg-ao-black/[0.79]">
+		<div className="absolute -z-50 h-screen bg-ao-white bg-gradient-to-t from-[#10151c] to-[#354364] xl:from-transparent xl:to-transparent xl:dark:bg-ao-black/[0.79]">
 			<Navbar />
-			<BlobHome />
+			{/* <BlobHome /> */}
```
Comment (RhysSullivan) [inline packages/ui/src/components/home/Home.tsx:12]: delete coment

File: packages/ui/src/components/primitives/Message.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	args: { message: mockMessageWithDiscordAccount() },
+};
+
+export const OverflowLetters: Story = {
+	args: {
+		...Primary.args,
+		message: {
+			...mockMessageWithDiscordAccount(),
+			content:
+				'Helloooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo',
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/Message.stories.tsx:31]: this currently doesnt wrap on overflow, plz fix

File: packages/ui/src/components/primitives/Message.tsx [hunk 5]
```diff
@@ -0,0 +1,234 @@
+// This component file is based off - https://www.youtube.com/watch?v=vPRdY87_SH0
+import type {
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/Message.tsx:2]: Make sure messages work in light & dark theme

File: packages/ui/src/components/primitives/Navbar.tsx [hunk 6]
```diff
@@ -1,35 +1,66 @@
...
 import { useTheme } from '~ui/utils/index';
 import { AnswerOverflowLogo } from '../AnswerOverflowLogo';
 import { Button } from './Button';
 
 export const Navbar = () => {
+	const [navbarStyles, setNavbarStyles] = useState<string>('');
 	const theme = useTheme();
+	const navbarRef = useRef<HTMLDivElement>(null);
+
+	// // When the user scrolls down make the navbar sticky
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/Navbar.tsx:12]: delete unused comments

File: packages/ui/src/components/search/MessageResultDriver.tsx [hunk 7]
```diff
@@ -0,0 +1,8 @@
+//
+export const MessageResultDriver = () => {
```
Comment (RhysSullivan) [inline packages/ui/src/components/search/MessageResultDriver.tsx:2]: what is message result? if it is search results lets uniform that all over the place to only use search results

Message result page is the page you view after clicking a search result
Search results is the search results

File: packages/ui/src/components/search/SearchResult.tsx [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+							<ServerInviteJoinButton />
+						</>
+					)}
+				</div>
+				<SearchResultMetaData result={result} />
+			</div>
+		);
+	};
+
+	// const SearchResultMobileBar = () => {
```
Comment (RhysSullivan) [inline packages/ui/src/components/search/SearchResult.tsx:146]: unused commetns delete

File: packages/ui/src/components/search/SearchResult.tsx [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+	// 			</div>
+	// 		</div>
+	// 	);
+	// };
+
+	return (
+		<div className="flex h-full w-full flex-col rounded-standard bg-[#181B1F] lg:flex-row">
+			<SearchResultMainContent />
+			<SearchResultSidebar />
+			{/* <SearchResultMobileBar /> */}
```
Comment (RhysSullivan) [inline packages/ui/src/components/search/SearchResult.tsx:166]: unused comments delete

File: packages/ui/src/components/search/SearchResult.tsx [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+import {
+	Message,
+	MessageContents,
+	MessageImages,
+	MessageRenderer,
+	MessageTitle,
+} from '~ui/components/primitives/Message';
+import { Button } from '../primitives/Button';
+
+export interface MessageResultProps {
```
Comment (RhysSullivan) [inline packages/ui/src/components/search/SearchResult.tsx:11]: Can we make this look more like the message component which uses context?

File: packages/ui/src/components/search/SearchResult.tsx [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+// TODO: Use data from the API
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const SearchResultMetaData = ({ result }: MessageResultProps) => {
+	return (
+		<div className="mt-2 flex flex-row items-center justify-center">
+			{/* Chat */}
+			<span className="flex items-center justify-center px-1 text-white/[.55]">
+				{0}
+			</span>
+			<svg
```
Comment (RhysSullivan) [inline packages/ui/src/components/search/SearchResult.tsx:54]: Make this SVG its own component, its hard to read w/ all this in here

**Review by RhysSullivan — CHANGES_REQUESTED** [review]
Good start, needs a few changes but has a good base

File: packages/config/eslint-config-custom/next.js (hunk 1 above)
Comment (JoltCode) [reply packages/config/eslint-config-custom/next.js:26]: Reverted

File: packages/ui/src/components/search/SearchResult.tsx (hunk 11 above)
Comment (JoltCode) [reply packages/ui/src/components/search/SearchResult.tsx]: Done

File: packages/ui/src/components/primitives/Navbar.tsx (hunk 6 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/Navbar.tsx]: Done

File: packages/ui/src/components/primitives/Message.stories.tsx (hunk 4 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/Message.stories.tsx:31]: Fixed 👍

File: packages/ui/src/components/home/Home.tsx (hunk 2 above)
Comment (JoltCode) [reply packages/ui/src/components/home/Home.tsx]: Done

File: packages/ui/src/components/search/SearchResult.tsx (hunk 10 above)
Comment (JoltCode) [reply packages/ui/src/components/search/SearchResult.tsx]: Done

File: packages/ui/src/components/primitives/Message.tsx (hunk 5 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/Message.tsx:2]: Done! 👍

**Review by RhysSullivan — CHANGES_REQUESTED** [review]
Looks good! One minor change and then thats ready to merge and I'll get it all hooked up

**Comment by RhysSullivan** [comment]
Also we need CI passing, the new formatting command is yarn format

**Comment by RhysSullivan** [comment]
We need the "jump to message" / some way to view the result, it needs to a <Link/> with an href to the question message id

File: packages/ui/src/components/primitives/Input.tsx [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+						placeholder={props.placeholder}
+						fill={props.fill}
+					/>
+					<InputButton
+						buttonAria={props.buttonAria}
+						onChange={function (text: string): void {
+							throw new Error('Function not implemented.' + text);
+						}}
+					>
+						<svg
```
[the remark this answers is not in the document]
Comment (JoltCode) [reply packages/ui/src/components/primitives/Input.tsx:67]: Oops! Will do 👍

**Comment by JoltCode** [comment]
@RhysSullivan would you mind sorting the errors please? Seems that the merge from main has gone weird or something....

**Comment by RhysSullivan** [comment]
Will do

**Comment by JoltCode** [comment]
Thanks!

**Pull request merged**